README.md
# bee-tree-select

[![npm version](https://img.shields.io/npm/v/bee-tree-select.svg)](https://www.npmjs.com/package/bee-tree-select)
[![Build Status](https://img.shields.io/travis/tinper-bee/bee-tree-select/master.svg)](https://travis-ci.org/tinper-bee/bee-tree-select)
[![Coverage Status](https://coveralls.io/repos/github/tinper-bee/bee-tree-select/badge.svg?branch=master)](https://coveralls.io/github/tinper-bee/bee-tree-select?branch=master)
[![devDependency Status](https://img.shields.io/david/dev/tinper-bee/bee-tree-select.svg)](https://david-dm.org/tinper-bee/bee-tree-select#info=devDependencies)
[![NPM downloads](http://img.shields.io/npm/dm/bee-tree-select.svg?style=flat)](https://npmjs.org/package/bee-tree-select)
[![Average time to resolve an issue](http://isitmaintained.com/badge/resolution/tinper-bee/bee-tree-select.svg)](http://isitmaintained.com/project/tinper-bee/bee-tree-select "Average time to resolve an issue")
[![Percentage of issues still open](http://isitmaintained.com/badge/open/tinper-bee/bee-tree-select.svg)](http://isitmaintained.com/project/tinper-bee/bee-tree-select "Percentage of issues still open")

bee-tree-select 是树型选择控件，当select下拉框选项为树形结构时会用到该组件。例如层级之间的产品、学科系统、分类目录等等。


## 依赖

- react >= 15.3.0
- react-dom >= 15.3.0
- prop-types >= 15.6.0

## 使用方法

#### 组件引入
先进行下载TreeSelect包
```
npm install --save bee-tree-select
```
组件调用
```js
import Tree from 'bee-tree-select';
import React, { Component } from 'react';
import TreeSelect,{TreeNode} from 'bee-tree-select';
class Demo1 extends Component {
    state = {
    value: undefined,
  }
  onChange = (value) => {
    console.log(arguments);
    this.setState({ value });
  }
  render () {
    return (

            <TreeSelect
            showSearch
            style={{ width: 300 }}
            value={this.state.value}
            dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}
            placeholder="Please select"
            allowClear
            treeDefaultExpandAll
            onChange={this.onChange}
            >
                <TreeNode value="parent 1" title="parent 1" key="0-1">
                  <TreeNode value="parent 1-0" title="parent 1-0" key="0-1-1">
                    <TreeNode value="leaf1" title="my leaf" key="random" />
                    <TreeNode value="leaf2" title="your leaf" key="random1" />
                  </TreeNode>
                  <TreeNode value="parent 1-1" title="parent 1-1" key="random2">
                    <TreeNode value="sss" key="random3" />
                  </TreeNode>
                </TreeNode>
            </TreeSelect>

    )
  }
}
```
#### 样式引入
- 可以使用link引入dist目录下tree.css
```
<link rel="stylesheet" href="./node_modules/build/bee-tree-select.css">
```
- 可以在js中import样式
```js
import "./node_modules/src/TreeSelect.scss"
//或是
import "./node_modules/build/bee-tree-select.css"
```

## API

### Tree props

| 参数 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| allowClear | 显示清除按钮 | boolean | false |
| className | 自定义输入框样式名 | string | - |
| defaultValue | 指定默认选中的条目 | string/string\[] | - |
| disabled | 是否禁用 | boolean | false |
| dropdownClassName | 下拉菜单的 className 属性 | string | - |
| dropdownMatchSelectWidth | 下拉菜单和选择器同宽 | boolean | true |
| dropdownStyle | 下拉菜单的样式 | object | - |
| notFoundContent | 设定搜索不到数据显示的内容 | String | '无匹配结果' |
| filterTreeNode | 是否根据输入项进行筛选，默认用 treeNodeFilterProp 的值作为要筛选的 TreeNode 的属性值 | boolean\|Function(inputValue: string, treeNode: TreeNode) (函数需要返回bool值) | Function |
| getPopupContainer | 菜单渲染父节点。默认渲染到 body 上，如果你遇到菜单滚动定位问题，试试修改为滚动的区域，并相对其定位。 | Function(triggerNode) | () => document.body |
| labelInValue | 是否把每个选项的 label 包装到 value 中，会把 value 类型从 `string` 变为 `{value: string, label: ReactNode, halfChecked(treeCheckStrictly 时有效): string[] }` 的格式 | boolean | false |
| loadData | 异步加载数据 | function(node) | - |
| multiple | 支持多选（当设置 treeCheckable 时自动变为true） | boolean | false |
| placeholder | 选择框默认文字 | string | - |
| searchPlaceholder | 搜索框默认文字 | string | - |
| showCheckedStrategy | 定义选中项回填的方式。`TreeSelect.SHOW_ALL`: 显示所有选中节点(包括父节点). `TreeSelect.SHOW_PARENT`: 只显示父节点(当父节点下所有子节点都选中时). 默认只显示子节点. | enum{TreeSelect.SHOW_ALL, TreeSelect.SHOW_PARENT, TreeSelect.SHOW_CHILD } | TreeSelect.SHOW_CHILD |
| showSearch | 在下拉中显示搜索框(仅在单选模式下生效) | boolean | false |
| treeCheckable | 显示 checkbox | boolean | false |
| treeCheckStrictly | checkable 状态下节点选择完全受控（父子节点选中状态不再关联），会使得 `labelInValue` 强制为 true | boolean | false |
| treeData | treeNodes 数据，如果设置则不需要手动构造 TreeNode 节点（value 在整个树范围内唯一） | array&lt;{value, label, children, [disabled, disableCheckbox, selectable]}>(如果定义了title，label会被title覆盖) | \[] |
| treeDataSimpleMode | 使用简单格式的 treeData，具体设置参考可设置的类型 (此时 treeData 应变为这样的数据结构: [{id:1, pId:0, value:'1', label:"test1",...},...], `pId` 是父节点的 id) | false\|Array&lt;{ id: string, pId: string, rootPId: null }> | false |
| treeDefaultExpandAll | 默认展开所有树节点 | boolean | false |
| treeDefaultExpandedKeys | 默认展开的树节点 | string\[] | - |
| treeNodeFilterProp | 输入项过滤对应的 treeNode 属性 | string | 'value' |
| treeNodeLabelProp | 作为显示的 prop 设置 | string | 'title' |
| value | 指定当前选中的条目 | string/string\[] | - |
| onChange | 选中树节点时调用此函数 | function(value, label, extra) | - |
| onSearch | 文本框值变化时回调 | function(value: string) | - |
| onSelect | 被选中时调用 | function(value, node, extra) | - |


### TreeNode props

> 建议使用 treeData 来代替 TreeNode，免去手工构造麻烦

| 参数 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| disableCheckbox | 禁掉 checkbox | boolean | false |
| disabled | 是否禁用 | boolean | false |
| isLeaf | 是否是叶子节点 | boolean | false |
| key | 此项必须设置（其值在整个树范围内唯一） | string | - |
| title | 树节点显示的内容 | string\|ReactNode | '---' |
| value | 默认根据此属性值进行筛选（其值在整个树范围内唯一） | string | - |
#### 开发调试

```sh
$ npm install -g bee-tools
$ git clone https://github.com/tinper-bee/bee-tree-select
$ cd bee-tree-select
$ npm install
$ npm run dev
```


CHANGELOG.md
## [0.0.9](https://github.com/tinper-bee/bee-tree-select/compare/v0.0.8...v0.0.9) (2021-08-18)



## [0.0.8](https://github.com/tinper-bee/bee-tree-select/compare/v0.0.7...v0.0.8) (2021-08-18)


### Bug Fixes

* 可以传真正的searchPlaceholder ([135c928](https://github.com/tinper-bee/bee-tree-select/commit/135c928312d8e4eba3c0390a6a447668a5382203))



## [0.0.7](https://github.com/tinper-bee/bee-tree-select/compare/v0.0.6...v0.0.7) (2020-04-24)


### Bug Fixes

* 自动化 ([31f32c5](https://github.com/tinper-bee/bee-tree-select/commit/31f32c53d68d5f8c449b6c85f9bc2fa779f08b69))



<a name="0.0.6"></a>
## [0.0.6](https://github.com/tinper-bee/bee-tree-select/compare/v0.0.5...v0.0.6) (2019-09-02)



<a name="0.0.5"></a>
## [0.0.5](https://github.com/tinper-bee/bee-tree-select/compare/v0.0.4...v0.0.5) (2019-06-01)



<a name="0.0.4"></a>
## [0.0.4](https://github.com/tinper-bee/bee-tree-select/compare/v0.0.3...v0.0.4) (2019-06-01)



<a name="0.0.3"></a>
## [0.0.3](https://github.com/tinper-bee/bee-tree-select/compare/v0.0.2...v0.0.3) (2019-05-24)



<a name="0.0.2"></a>
## 0.0.2 (2018-05-23)





demo/index.js
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Con, Row, Col } from 'bee-layout';
import { Panel } from 'bee-panel';
import Drawer from 'bee-drawer';
import Clipboard from 'bee-clipboard'; 



var Demo1 = require("./demolist/Demo1");var Demo2 = require("./demolist/Demo2");var Demo3 = require("./demolist/Demo3");var Demo4 = require("./demolist/Demo4");var Demo5 = require("./demolist/Demo5");var DemoArray = [{"example":<Demo1 />,"title":" treeSelect基本使用","code":"/**\n*\n* @title treeSelect基本使用\n* @description treeSelect基本使用—单选\n*\n*/\nimport React, { Component } from 'react';\nimport TreeSelect,{TreeNode} from '../../src';\nclass Demo1 extends Component {\n    state = {\n    value: undefined,\n  }\n\n  onChange = (value) => {\n    this.setState({ value });\n  }\n  onSelect = (value, node, extra) =>{\n    console.log('--value--'+value);\n  }\n\n  render () {\n    return (\n      <TreeSelect\n      showSearch\n      style={{ width: 300 }}\n      value={this.state.value}\n      dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}\n      placeholder=\"请选择\"\n      allowClear\n      treeDefaultExpandAll\n      onChange={this.onChange}\n      onSelect={this.onSelect}\n      >\n        <TreeNode value=\"parent 1\" title=\"parent 1\" key=\"0-1\">\n          <TreeNode value=\"parent 1-0\" title=\"parent 1-0\" key=\"0-1-1\">\n            <TreeNode value=\"leaf1\" title=\"my leaf\" key=\"random\" />\n            <TreeNode value=\"leaf2\" title=\"your leaf\" key=\"random1\" />\n          </TreeNode>\n          <TreeNode value=\"parent 1-1\" title=\"parent 1-1\" key=\"random2\">\n            <TreeNode value=\"sss\" title=\"sss\" key=\"random3\" />\n          </TreeNode>\n        </TreeNode>\n      </TreeSelect>\n    )\n  }\n}\nexport default Demo1","desc":" treeSelect基本使用—单选"},{"example":<Demo2 />,"title":" treeSelect 使用JSON数据","code":"/**\n*\n* @title treeSelect 使用JSON数据\n* @description 使用 treeData 把 JSON 数据直接生成树结构。\n*\n*/\nimport React, { Component } from 'react';\nimport { TreeSelect } from 'tinper-bee';\n\n\nconst treeData = [{\n  title: <span title=\"root 1\">parent 1</span>,\n  value: '0-0',\n  key: '0-0',\n  children: [{\n    title: 'Child Node1',\n    value: '0-0-1',\n    key: '0-0-1',\n  }, {\n    title: 'Child Node2',\n    value: '0-0-2',\n    key: '0-0-2',\n  }],\n}, {\n  title: 'Node2',\n  value: '0-1',\n  key: '0-1',\n}];\n\n\nclass Demo2 extends Component {\n    state = {\n    value: undefined,\n  }\n  onChange = (value) => {\n    this.setState({ value });\n  }\n  render () {\n    return (\n\n              <TreeSelect\n                  style={{ width: 300 }}\n                  value={this.state.value}\n                  dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}\n                  treeData={treeData}\n                  placeholder=\"Please select\"\n                  treeDefaultExpandAll\n                  onChange={this.onChange}\n              />\n\n    )\n  }\n}\nexport default Demo2","desc":" 使用 treeData 把 JSON 数据直接生成树结构。"},{"example":<Demo3 />,"title":" treeSelect 多选","code":"/**\n*\n* @title treeSelect 多选\n* @description 添加multiple属性，下拉树选项可以多选\n*\n*/\nimport React, { Component } from 'react';\nimport TreeSelect,{TreeNode} from '../../src';\n\nconst treeData = [{\n  title: '0-0',\n  value: '0-0',\n  key: '0-0',\n  children: [{\n    title: '0-0-1',\n    value: '0-0-1',\n    key: '0-0-1',\n  }, {\n    title: '0-0-2',\n    value: '0-0-2',\n    key: '0-0-2',\n  }],\n}, {\n  title: '0-1',\n  value: '0-1',\n  key: '0-1',\n}];\n\n\nclass Demo3 extends Component {\n    state = {\n    value: undefined,\n  }\n  onChange = (value) => {\n    this.setState({ value });\n  }\n  render () {\n    return (\n\n        <TreeSelect\n            style={{ width: 300 }}\n            value={this.state.value}\n            dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}\n            treeData={treeData}\n            multiple\n            treeDefaultExpandAll\n            onChange={this.onChange}\n            showSearch\n        />\n\n    )\n  }\n}\nexport default Demo3\n","desc":" 添加multiple属性，下拉树选项可以多选"},{"example":<Demo4 />,"title":" treeSelect 勾选框树","code":"/**\n*\n* @title treeSelect 勾选框树\n* @description 使用勾选框实现多选功能。\n*\n*/\nimport React, { Component } from 'react';\nimport TreeSelect,{TreeNode} from '../../src';\n\nconst SHOW_PARENT = TreeSelect.SHOW_PARENT;\n\nconst treeData = [{\n  title: '0-0',\n  value: '0-0',\n  key: '0-0',\n  children: [{\n    title: '0-0-1',\n    value: '0-0-1',\n    key: '0-0-1',\n  }, {\n    title: '0-0-2',\n    value: '0-0-2',\n    key: '0-0-2',\n  }],\n}, {\n  title: '0-1',\n  value: '0-1',\n  key: '0-1',\n}];\n\n\nclass Demo4 extends Component {\n    state = {\n    value: undefined,\n  }\n  onChange = (value) => {\n    this.setState({ value });\n  }\n  render () {\n    return (\n\n              <TreeSelect\n                style={{ width: 300 }}\n                value={this.state.value}\n                dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}\n                treeData={treeData}\n                treeCheckable= {true}\n                showCheckedStrategy={SHOW_PARENT}\n                treeDefaultExpandAll\n                onChange={this.onChange}\n                showSearch\n            />\n\n    )\n  }\n}\nexport default Demo4\n","desc":" 使用勾选框实现多选功能。"},{"example":<Demo5 />,"title":" treeSelect 动态加载树","code":"/**\n*\n* @title treeSelect 动态加载树\n* @description 点击父节点动态加载树\n*\n*/\nimport React, { Component } from 'react';\nimport TreeSelect,{TreeNode} from '../../src';\nimport { getNewTreeData, generateTreeNodes } from './util';\n\n\nclass Demo4 extends Component {\n    static propTypes = {};\n\n    state = {\n      treeData: [\n        { title: 'pNode 01', value: '0-0', key: '0-0' },\n        { title: 'pNode 02', value: '0-1', key: '0-1' },\n        { title: 'pNode 03', value: '0-2', key: '0-2', isLeaf: true },\n      ],\n      // value: '0-0',\n      value: { value: '0-0-0-value', title: '0-0-0-title' },\n    };\n  \n    onChange = (value) => {\n      console.log(value);\n      this.setState({\n        value,\n      });\n    };\n  \n    onLoadData = (treeNode) => {\n      console.log(treeNode);\n      return new Promise((resolve) => {\n        setTimeout(() => {\n          const treeData = [...this.state.treeData];\n          getNewTreeData(treeData, treeNode.props.eventKey, generateTreeNodes(treeNode), 2);\n          this.setState({ treeData });\n          resolve();\n        }, 500);\n      });\n    };\n  render () {\n    return (\n\n        <TreeSelect\n        style={{ width: 300 }}\n        treeData={this.state.treeData}\n        labelInValue\n        value={this.state.value}\n        onChange={this.onChange}\n        loadData={this.onLoadData}\n      />\n\n    )\n  }\n}\nexport default Demo4","desc":" 点击父节点动态加载树"}]


class Demo extends Component {
    constructor(props){
        super(props);
        this.state = {
            open: false
        }
    }
    handleClick=()=> {
        this.setState({ open: !this.state.open })
    }
    fCloseDrawer=()=>{
        this.setState({
            open: false
        })
    }

    render () {
        const { title, example, code, desc, scss_code  } = this.props;

        const header = (
            <div>
                <p className='component-title'>{ title }</p>
                <p>{ desc }</p>
                <span className='component-code' onClick={this.handleClick}> 查看源码 <i className='uf uf-arrow-right'/> </span>
            </div>
        );
        return (
            <Col md={12} id={title.trim()} className='component-demo'>
            <Panel header={header}>
                {example}
            </Panel>
           
            <Drawer className='component-drawerc' title={title} show={this.state.open} placement='right' onClose={this.fCloseDrawer}>
            <div className='component-code-copy'> JS代码 
                <Clipboard action="copy" text={code}/>
            </div>
            <pre className="pre-js">
                <code className="hljs javascript">{ code }</code>
            </pre >
            {!!scss_code ?<div className='component-code-copy copy-css'> SCSS代码 
                <Clipboard action="copy" text={scss_code}/>
            </div>:null }
                { !!scss_code ? <pre className="pre-css">
                 <code className="hljs css">{ scss_code }</code>
                 </pre> : null }
            </Drawer>
        </Col>
    )
    }
}

class DemoGroup extends Component {
    constructor(props){
        super(props)
    }
    render () {
        return (
            <Row>
            {DemoArray.map((child,index) => {

                return (
            <Demo example= {child.example} title= {child.title} code= {child.code} scss_code= {child.scss_code} desc= {child.desc} key= {index}/>
    )

    })}
    </Row>
    )
    }
}

ReactDOM.render(<DemoGroup/>, document.getElementById('tinperBeeDemo'));


src/TreeSelect.js
/**
* This source code is quoted from rc-tree-select.
* homepage: https://github.com/react-component/tree-select
*/
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import RcTreeSelect,{TreeNode,SHOW_ALL, SHOW_PARENT, SHOW_CHILD} from 'rc-tree-select';
import Icon from 'bee-icon';
import classNames from 'classnames';
import warning from 'warning';
import omit from 'omit.js';

const defaultProps =   {
    prefixCls: 'u-select',
    transitionName: 'slide-up',
    choiceTransitionName: 'zoom',
    showSearch: false,
    notFoundContent:"无匹配结果",
    searchPlaceholder: '请搜索'
};
const propTypes = {
    getPopupContainer: PropTypes.func,
};

class TreeSelect extends Component {
    static TreeNode = TreeNode;
    static SHOW_ALL = SHOW_ALL;
    static SHOW_PARENT = SHOW_PARENT;
    static SHOW_CHILD = SHOW_CHILD;

    constructor(props) {
        super(props);

        warning(
          props.multiple !== false || !props.treeCheckable,
          'TreeSelect',
          '`multiple` will alway be `true` when `treeCheckable` is true',
        );
    }
    focus() {
        this.rcTreeSelect.focus();
    }

    blur() {
        this.rcTreeSelect.blur();
    }

    saveTreeSelect = (node) => {
        this.rcTreeSelect = node;
    };

    renderSwitcherIcon = (prefixCls, { isLeaf, loading }) => {
        if (loading) {
          return <Icon type="loading" className={`${prefixCls}-switcher-loading-icon`} />;
        }
        if (isLeaf) {
          return null;
        }
        return <Icon type="uf-triangle-down" className={`${prefixCls}-switcher-icon`} />;
    };

    render(){
        const {
            prefixCls,
            className,
            size,
            notFoundContent,
            dropdownStyle,
            dropdownClassName,
            suffixIcon,
            getPopupContainer,
            searchPlaceholder,
            ...restProps,
        } = this.props;
        const rest = omit(restProps, ['inputIcon', 'removeIcon', 'clearIcon', 'switcherIcon']);

        const cls = classNames(
            {
                [`${prefixCls}-lg`]: size === 'large',
                [`${prefixCls}-sm`]: size === 'small',
            },
            className,
        );

        // showSearch: single - false, multiple - true
        let { showSearch } = restProps;
        if (!('showSearch' in restProps)) {
            showSearch = !!(restProps.multiple || restProps.treeCheckable);
        }

        let checkable = rest.treeCheckable;
        if (checkable) {
            checkable = <span className={`${prefixCls}-tree-checkbox-inner`} />;
        }

        const inputIcon = (suffixIcon &&
        (React.isValidElement(suffixIcon)
            ? React.cloneElement(suffixIcon)
            : suffixIcon)) || <Icon type="uf-treearrow-down" className={`${prefixCls}-arrow-icon`} />;

        const removeIcon = <Icon type="uf-close" className={`${prefixCls}-remove-icon`} />;

        const clearIcon = (
            <Icon type="uf-close-c" className={`${prefixCls}-clear-icon`} />
        );

        return (
            <RcTreeSelect
                // switcherIcon={(nodeProps) =>
                //     this.renderSwitcherIcon(prefixCls, nodeProps)
                // }
                inputIcon={inputIcon}
                removeIcon={removeIcon}
                clearIcon={clearIcon}
                {...restProps}
                showSearch={showSearch}
                getPopupContainer={getPopupContainer}
                dropdownClassName={classNames(dropdownClassName, `${prefixCls}-dropdown`)}
                prefixCls={prefixCls}
                className={cls}
                dropdownStyle={{ maxHeight: '100vh', overflow: 'auto', ...dropdownStyle }}
                notFoundContent={notFoundContent}
                treeCheckable={checkable}
                ref={this.saveTreeSelect}
                searchPlaceholder={searchPlaceholder}
            />
        )
    }

};
TreeSelect.propTypes = propTypes;
TreeSelect.defaultProps = defaultProps;
export default TreeSelect;


demo/demolist/Demo2.js
/**
*
* @title treeSelect 使用JSON数据
* @description 使用 treeData 把 JSON 数据直接生成树结构。
*
*/
import React, { Component } from 'react';
import TreeSelect from '../../src';

const treeData = [{
  title: <span title="root 1">parent 1</span>,
  value: '0-0',
  key: '0-0',
  children: [{
    title: 'Child Node1',
    value: '0-0-1',
    key: '0-0-1',
  }, {
    title: 'Child Node2',
    value: '0-0-2',
    key: '0-0-2',
  }],
}, {
  title: 'Node2',
  value: '0-1',
  key: '0-1',
}];


class Demo2 extends Component {
    state = {
    value: undefined,
  }
  onChange = (value) => {
    this.setState({ value });
  }
  render () {
    return (

              <TreeSelect
                  style={{ width: 300 }}
                  value={this.state.value}
                  dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}
                  treeData={treeData}
                  placeholder="Please select"
                  treeDefaultExpandAll
                  onChange={this.onChange}
              />

    )
  }
}
export default Demo2

demo/demolist/Demo3.js
/**
*
* @title treeSelect 多选
* @description 添加multiple属性，下拉树选项可以多选
*
*/
import React, { Component } from 'react';
import TreeSelect,{TreeNode} from '../../src';

const treeData = [{
  label: 'Node1',
  value: '0-0',
  key: '0-0',
  children: [{
    label: 'Child Node1',
    value: '0-0-1',
    key: '0-0-1',
  }, {
    label: 'Child Node2',
    value: '0-0-2',
    key: '0-0-2',
  }],
}, {
  label: 'Node2',
  value: '0-1',
  key: '0-1',
}];


class Demo3 extends Component {
    state = {
    value: undefined,
  }
  onChange = (value) => {
    this.setState({ value });
  }
  render () {
    return (

              <TreeSelect
                  style={{ width: 300 }}
                  value={this.state.value}
                  dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}
                  treeData={treeData}
                    multiple
                  treeDefaultExpandAll
                  onChange={this.onChange}
              />

    )
  }
}
export default Demo3

demo/demolist/Demo4.js
/**
*
* @title treeSelect 勾选框树
* @description 使用勾选框实现多选功能。
*
*/
import React, { Component } from 'react';
import TreeSelect,{TreeNode} from '../../src';

const SHOW_PARENT = TreeSelect.SHOW_PARENT;

const treeData = [{
  label: 'Node1',
  value: '0-0',
  key: '0-0',
  children: [{
    label: 'Child Node1',
    value: '0-0-1',
    key: '0-0-1',
  }, {
    label: 'Child Node2',
    value: '0-0-2',
    key: '0-0-2',
  }],
}, {
  label: 'Node2',
  value: '0-1',
  key: '0-1',
}];


class Demo4 extends Component {
    state = {
    value: undefined,
  }
  onChange = (value) => {
    this.setState({ value });
  }
  render () {
    return (

              <TreeSelect
                style={{ width: 300 }}
                value={this.state.value}
                dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}
                treeData={treeData}
                treeCheckable= {true}
                showCheckedStrategy={SHOW_PARENT}
                treeDefaultExpandAll
                onChange={this.onChange}
            />

    )
  }
}
export default Demo4

demo/demolist/Demo5.js
/**
*
* @title treeSelect 动态加载树
* @description 点击父节点动态加载树
*
*/
import React, { Component } from 'react';
import TreeSelect,{TreeNode} from '../../src';
import { getNewTreeData, generateTreeNodes } from './util';


class Demo4 extends Component {
    static propTypes = {};

    state = {
      treeData: [
        { label: 'pNode 01', value: '0-0', key: '0-0' },
        { label: 'pNode 02', value: '0-1', key: '0-1' },
        { label: 'pNode 03', value: '0-2', key: '0-2', isLeaf: true },
      ],
      // value: '0-0',
      value: { value: '0-0-0-value', label: '0-0-0-label' },
    };
  
    onChange = (value) => {
      console.log(value);
      this.setState({
        value,
      });
    };
  
    onLoadData = (treeNode) => {
      console.log(treeNode);
      return new Promise((resolve) => {
        setTimeout(() => {
          const treeData = [...this.state.treeData];
          getNewTreeData(treeData, treeNode.props.eventKey, generateTreeNodes(treeNode), 2);
          this.setState({ treeData });
          resolve();
        }, 500);
      });
    };
  render () {
    return (

        <TreeSelect
        style={{ width: 300 }}
        treeData={this.state.treeData}
        labelInValue
        value={this.state.value}
        onChange={this.onChange}
        loadData={this.onLoadData}
      />

    )
  }
}
export default Demo4

demo/index-demo-base.js

import { Con, Row, Col } from 'bee-layout';
import { Panel } from 'bee-panel';
import Button from 'bee-button';
import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import TreeSelect from '../src';


const CARET = <i className="uf uf-arrow-down"></i>;

const CARETUP = <i className="uf uf-arrow-up"></i>;


{demolist}

class Demo extends Component {
    constructor(props){
        super(props);
        this.state = {
            open: false
        }
        this.handleClick = this.handleClick.bind(this);
    }
    handleClick() {
        this.setState({ open: !this.state.open })
    }

    render () {
        const { title, example, code, desc  } = this.props;
        let caret = this.state.open ? CARETUP : CARET;
        let text = this.state.open ? "隐藏代码" : "查看代码";

        const footer = (
            <Button shape="block" onClick={ this.handleClick }>
                { caret }
                { text }
            </Button>
        );
        return (
            <Col md={12} >
                <h3>{ title }</h3>
                <p>{ desc }</p>
                <Panel collapsible headerContent expanded={ this.state.open } colors='bordered' header={ example } footer={footer} footerStyle = {{padding: 0}}>
                    <pre><code className="hljs javascript">{ code }</code></pre>
                </Panel>
            </Col>
        )
    }
}

class DemoGroup extends Component {
    constructor(props){
        super(props)
    }
    render () {
        return (
                <Row>
                    {DemoArray.map((child,index) => {

                        return (
                            <Demo example= {child.example} title= {child.title} code= {child.code} desc= {child.desc} key= {index}/>
                        )

                    })}
                </Row>
        )
    }
}

ReactDOM.render(<DemoGroup/>, document.getElementById('tinperBeeDemo'));


demo/demolist/Demo1.js
/**
*
* @title treeSelect基本使用
* @description treeSelect基本使用—单选
*
*/
import React, { Component } from 'react';
import TreeSelect,{TreeNode} from '../../src';
class Demo1 extends Component {
    state = {
    value: undefined,
  }
  onChange = (value) => {
    this.setState({ value });
  }
  onSelect = (value, node, extra) =>{
    console.log('--value--'+value);
  }
  render () {
    return (

            <TreeSelect
            showSearch
            style={{ width: 300 }}
            value={this.state.value}
            dropdownStyle={{ maxHeight: 400, overflow: 'auto' }}
            placeholder="Please select"
            allowClear
            treeDefaultExpandAll
           size='small'
            onChange={this.onChange}
            onSelect={this.onSelect}
            >
                <TreeNode value="parent 1" title="parent 1" key="0-1">
                  <TreeNode value="parent 1-0" title="parent 1-0" key="0-1-1">
                    <TreeNode value="leaf1" title="my leaf" key="random" />
                    <TreeNode value="leaf2" title="your leaf" key="random1" />
                  </TreeNode>
                  <TreeNode value="parent 1-1" title="parent 1-1" key="random2">
                    <TreeNode value="sss" title="sss" key="random3" />
                  </TreeNode>
                </TreeNode>
            </TreeSelect>

    )
  }
}
export default Demo1